app/main.py

```diff
@@ -1,23 +1,39 @@
 import init_django_orm  # noqa: F401
 
-from django.db.models import QuerySet
+from django.db.models import QuerySet, Q, F
+
+from db.models import LoyaltyProgram, Customer, LoyaltyProgramParticipant
 
 
 def all_loyalty_program_names() -> QuerySet:
-    pass
+    return LoyaltyProgram.objects.values_list(
+        "name", "bonus_percentage"
+    )
 
 
 def not_active_customers() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects.exclude(
+        last_activity__gte="2022-01-01").filter(
+        last_activity__gte="2021-01-01").values(
```

Comment:
why didn't you filter in a simpler way?
`filter(last_activity__year=2021)`

Reply:
now i will filter using this way :)